src/backend/joanie/core/models/products.py

```diff
@@ -957,17 +958,39 @@ def submit_for_signature(self, user: User):
             name=contract_definition.name, context=context
         )
 
-        was_already_submitted = (
+        was_already_submitted = bool(
             contract.submitted_for_signature_on and contract.signature_backend_reference
         )
-        should_be_resubmitted = was_already_submitted and (
+
+        should_be_resubmitted = was_already_submitted and bool(
             not contract.is_eligible_for_signing() or contract.context != context
         )
 
         if should_be_resubmitted:
-            backend_signature.delete_signing_procedure(
-                contract.signature_backend_reference
-            )
+            # The signature provider may take some time to respond with the confirmation of the
+            # deletion. If we reach the timeout limit, we should reset the contract submission
+            # values because the signature provider will delete them.
+            # We won't be able to use the `contract.signature_backend_reference` again.
+            try:
+                backend_signature.delete_signing_procedure(
+                    contract.signature_backend_reference
+                )
+            except ReadTimeout as exception:  # pylint: disable=unused-variable
```

Comment:
I find that catching the request timeout exception here is a bit out of place. What do you think of catching it inside the `delete_signing_procedure`, right where the `delete` is done, and raising a custom exception?

Reply:
Thanks ! let's go for the custom exception.